src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -826,30 +811,21 @@ void SmartAI::SetCombatMove(bool on)
 {
     if (mCanCombatMove == on)
         return;
+
     mCanCombatMove = on;
+
     if (!IsAIControlled())
         return;
-    if (!HasEscortState(SMART_ESCORT_ESCORTING))
+
+    if (me->IsEngaged())
```

Comment:
@Treeston insight on this one? IsEngaged will mean that the creature is actually in proper combat? will react passive creatures ever "engage"?